src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php

```diff
@@ -53,6 +67,24 @@ public function validateForm(FormEvent $event)
 
                 $this->violationMapper->mapViolation($violation, $form, $allowNonSynchronized);
             }
+
+            $this->dispatchEvents(ValidatorFormEvents::POST_VALIDATE);
+        }
+    }
+
+    private function dispatchEvents(string $eventName)
+    {
+        if (!isset(self::$dispatchEvents[$eventName])) {
+            return;
         }
+
+        $event = array_flip(ValidatorFormEvents::ALIASES)[$eventName];
+
+        foreach (self::$dispatchEvents[$eventName] as $form) {
+            $event = new $event($form, $form->getData());
+            $form->getConfig()->getEventDispatcher()->dispatch($event, $eventName);
```

Comment:
can't `$form->getConfig()->getEventDispatcher()` be defined in a var before `foreach` ?

Reply:
`$form` is the result of the array we are looping on

Reply:
Oki got it,